src/fast_check/transform_dts.rs

```diff
@@ -0,0 +1,1509 @@
+      }
+    }
+  }
+
+  fn transform_ts_ns_body(&mut self, ns: TsNamespaceBody) -> TsNamespaceBody {
+    match ns {
+      TsNamespaceBody::TsModuleBlock(mut ts_module_block) => {
+        ts_module_block.body =
+          self.transform_module_items(ts_module_block.body);
+        TsNamespaceBody::TsModuleBlock(ts_module_block)
+      }
+      TsNamespaceBody::TsNamespaceDecl(ts_ns) => {
+        self.transform_ts_ns_body(*ts_ns.body)
+      }
+    }
+  }
+
+  // Support for expressions is limited in enums,
+  // see https://www.typescriptlang.org/docs/handbook/enums.html
+  fn valid_enum_init_expr(&mut self, expr: Expr) -> bool {
+    match expr {
+      Expr::Bin(bin_expr) => {
+        if !self.valid_enum_init_expr(*bin_expr.left) {
+          false
+        } else {
+          self.valid_enum_init_expr(*bin_expr.right)
+        }
+      }
+
+      Expr::Member(member_expr) => self.valid_enum_init_expr(*member_expr.obj),
+      Expr::OptChain(opt_expr) => match *opt_expr.base {
+        OptChainBase::Member(member_expr) => {
+          self.valid_enum_init_expr(Expr::Member(member_expr))
+        }
+        OptChainBase::Call(_) => false,
+      },
+      // TS does infer the type of identifiers
+      Expr::Ident(_) => true,
+      Expr::Lit(lit) => match lit {
+        Lit::Num(_) | Lit::Str(_) => true,
+        Lit::Bool(_)
+        | Lit::Null(_)
+        | Lit::BigInt(_)
+        | Lit::Regex(_)
+        | Lit::JSXText(_) => false,
+      },
+      Expr::Tpl(tpl_expr) => {
+        for expr in tpl_expr.exprs {
+          if !self.valid_enum_init_expr(*expr) {
+            return false;
+          }
+        }
+        true
+      }
+
+      Expr::Paren(paren_expr) => self.valid_enum_init_expr(*paren_expr.expr),
+
+      Expr::TsTypeAssertion(ts_ass) => {
+        // Only assertions to number are allowed for computed
+        // enum members.
+        match *ts_ass.type_ann {
+          TsType::TsLitType(ts_lit) => match ts_lit.lit {
+            TsLit::Number(_) => true,
+            TsLit::Str(_)
+            | TsLit::Bool(_)
+            | TsLit::BigInt(_)
+            | TsLit::Tpl(_) => false,
+          },
+          TsType::TsKeywordType(_)
+          | TsType::TsThisType(_)
+          | TsType::TsFnOrConstructorType(_)
+          | TsType::TsTypeRef(_)
+          | TsType::TsTypeQuery(_)
+          | TsType::TsTypeLit(_)
+          | TsType::TsArrayType(_)
+          | TsType::TsTupleType(_)
+          | TsType::TsOptionalType(_)
+          | TsType::TsRestType(_)
+          | TsType::TsUnionOrIntersectionType(_)
+          | TsType::TsConditionalType(_)
+          | TsType::TsInferType(_)
+          | TsType::TsParenthesizedType(_)
+          | TsType::TsTypeOperator(_)
+          | TsType::TsIndexedAccessType(_)
+          | TsType::TsMappedType(_)
+          | TsType::TsTypePredicate(_)
+          | TsType::TsImportType(_) => false,
+        }
+      }
+
+      Expr::TsAs(ts_as) => self.valid_enum_ts_type(*ts_as.type_ann),
```

Comment:
I get an error for this in declaration files

> In ambient enum declarations member initializer must be constant expression.ts(1066)

Reply:
This seems to work for a very small subset:
```
export enum Foo {
  A = 1 as number,
}
```